src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2444,13 +2462,12 @@ bool Creature::IsImmunedToSpellEffect(SpellInfo const* spellInfo, SpellEffectInf
     return Unit::IsImmunedToSpellEffect(spellInfo, spellEffectInfo, caster, requireImmunityPurgesEffectAttribute);
 }
 
-bool Creature::isElite() const
+bool Creature::IsElite() const
 {
     if (IsPet())
         return false;
 
-    uint32 rank = GetCreatureTemplate()->rank;
-    return rank != CREATURE_ELITE_NORMAL && rank != CREATURE_ELITE_RARE;
+    return !HasClassification(CreatureClassifications::Normal) && !HasClassification(CreatureClassifications::Rare) && !HasClassification(CreatureClassifications::MinusMob);
```

Comment:
This function should check allowed values instead of "not elite" ones

Reply:
I had kept reverse checking as the bool did originally, but I can update to what you suggested